pkgs/development/python-modules/m3u8/default.nix

```diff
@@ -1,23 +1,28 @@
-{ lib, buildPythonPackage, fetchFromGitHub, requests, iso8601, bottle, pytest, pytest-cov }:
+{ lib, buildPythonPackage, fetchFromGitHub, requests, iso8601, bottle, pytestCheckHook }:
 
 buildPythonPackage rec {
   pname = "m3u8";
-  version = "0.6.0";
+  version = "0.9.0";
 
   src = fetchFromGitHub {
     owner = "globocom";
     repo = pname;
     rev = version;
-    sha256 = "0cmg993icpsa1b19kljxvjwhs167bsqrs0ad4wnwsi8qq6na5d4p";
+    sha256 = "sha256-EfHhmV2otEgEy2OVohS+DF7dk97GFdWZ4cFCERZBmlA=";
   };
 
-  checkInputs = [ bottle pytest pytest-cov ];
+  propagatedBuildInputs = [ requests iso8601 ];
 
-  checkPhase = ''
-    pytest tests/test_{parser,model,variant_m3u8}.py
-  '';
+  checkInputs = [ bottle pytestCheckHook ];
 
-  propagatedBuildInputs = [ requests iso8601 ];
+  pytestFlagsArray = [
+    "tests/test_{parser,model,variant_m3u8}.py"
```

Comment:
Is this interpolation being interpreted by the shell or by pytest? If it's the shell it would be sensitive to e.g. `pytestCheckHook` changing the quotes used for it. Which is something that wasn't a worry when it was a `checkPhase`.

Reply:
It is expanded by the shell, so all three files are run:

``` sh
    args+=" ${pytestFlagsArray[@]}"
    eval "@pythonCheckInterpreter@ $args"
```

I thought that the idea was to prefer `pytestCheckHook`, that's why I changed it.

I can revert it, or list each file individually (I don't really know why I didn't do that).

Reply:
I changed it to:

``` nix
  pytestFlagsArray = [
    "tests/test_parser.py"
    "tests/test_model.py"
    "tests/test_variant_m3u8.py"
  ];
```

Makes more sense.